logger/analytics/endpointresolver.go

```diff
@@ -33,4 +33,6 @@ func environmentVariableEndpointResolver(service, region string, optFns ...func(
 	return endpoints.DefaultResolver().EndpointFor(service, region, optFns...)
 }
 
+// EndpointResolver is used to override the endpoints that AWS clients use.
+// In particular for reducing networking costs for cross-region traffic.
```

Comment:
Is this the correct understanding? Also iirc we set an env var for firehose for this in catapult, should we do the same for kinesis now too/

Reply:
IIRC it happens auto-magically in most cases--the only time we want to override it is if we want to use the VPC endpoint set up in a /different/ VPC, e.g. if we're hitting Kinesis in us-west-1 from code running in us-west-2, then the code in us-west-2 needs to be configured with the us-west-1 VPC endpoint URL, and then the traffic gets routed internally across vpc peering or transit gateway to us-west-1 to reach the VPC endpoint

Reply:
And the understanding is mostly correct, but with the biggest thing being we use it in cases where we want to hit a VPC endpoint in a different region. This is usually for cost reasons since we'd prefer for the traffic to travel over the internal network vs. out to the public internet through a NAT Gateway